packages/mui-material/src/MenuList/MenuList.js

```diff
@@ -232,6 +232,17 @@ const MenuList = React.forwardRef(function MenuList(props, ref) {
         activeItemIndex = index;
       }
     }
+
+    if (
+      activeItemIndex === index &&
+      (child.props.disabled || child.props.muiSkipListHighlight || child.type.muiSkipListHighlight)
+    ) {
+      activeItemIndex += 1;
+      if (activeItemIndex >= children.length) {
+        // there are no focusable items within the list.
+        activeItemIndex = -1;
+      }
+    }
```

Comment:
Hi @michaldudak 

I have been looking for solutions to skip items in the menu list. And it looks like this only skips the initial focus when the menu is opened via keyboard. The traversal logic doesn't respect the `muiSkipListHighlight` prop.

In other words, `muiSkipListHighlight` in MenuList will only skip the focusVisible state for the initial render, but it is ignored when the user press key up/down to traverse the list items.

Let me know if I should file an issue or how I can help further on fixing this. I might be missing the point of this prop too. So I'd be grateful if you can point out.

Thanks in advance!

Reply:
@ycmjason Answering on behalf of @michaldudak , the `muiSkipListHighlight` prop is an internal prop and should be used only if you wrap the `ListSubheader` in a custom component. We document it's use-case [here](https://mui.com/material-ui/react-select/#grouping).

As for skipping the item in a menu list, I guess you need to disable the item, then it should be able to skip the highlight.